File: ch13/python/connecting/table_storage_connect_example.py
# Python example: Connecting to Azure Table Storage using Connection String, SAS Token, and Managed Identity
# Requires: pip install azure-data-tables azure-identity

from azure.data.tables import TableServiceClient
from azure.core.credentials import AzureSasCredential
from azure.identity import DefaultAzureCredential

# 1. Connect using a connection string
connection_string = "<your_connection_string>"
table_name = "myTable"
service_client_conn_str = TableServiceClient.from_connection_string(conn_str=connection_string)
table_client_conn_str = service_client_conn_str.get_table_client(table_name)
print(f"Connected to Table Storage using connection string. Table: {table_name}")

# 2. Connect using a SAS token
storage_uri = "https://<your_account>.table.core.windows.net"
sas_token = "<your_sas_token>"
service_client_sas = TableServiceClient(endpoint=storage_uri, credential=AzureSasCredential(sas_token))
table_client_sas = service_client_sas.get_table_client(table_name)
print(f"Connected to Table Storage using SAS token. Table: {table_name}")

# 3. Connect using Managed Identity (DefaultAzureCredential)
# Make sure your environment supports Managed Identity (e.g., Azure VM, App Service, etc.)
service_client_msi = TableServiceClient(endpoint=storage_uri, credential=DefaultAzureCredential())
table_client_msi = service_client_msi.get_table_client(table_name)
print(f"Connected to Table Storage using Managed Identity. Table: {table_name}")

File: ch13/java/connecting/TableStorageConnectExample.java
// Java example: Connecting to Azure Table Storage using Connection String, SAS Token, and Managed Identity
// Requires Maven dependencies: com.azure:azure-data-tables, com.azure:azure-identity

import com.azure.data.tables.TableClient;
import com.azure.data.tables.TableClientBuilder;
import com.azure.data.tables.TableServiceClient;
import com.azure.data.tables.TableServiceClientBuilder;
import com.azure.core.credential.AzureSasCredential;
import com.azure.identity.DefaultAzureCredentialBuilder;

public class TableStorageConnectExample {
    public static void main(String[] args) {
        String tableName = "myTable";

        // 1. Connect using a connection string
        String connectionString = "<your_connection_string>";
        TableServiceClient serviceClientConnStr = new TableServiceClientBuilder()
                .connectionString(connectionString)
                .buildClient();
        TableClient tableClientConnStr = serviceClientConnStr.getTableClient(tableName);
        System.out.println("Connected to Table Storage using connection string. Table: " + tableName);

        // 2. Connect using a SAS token
        String storageUri = "https://<your_account>.table.core.windows.net";
        String sasToken = "<your_sas_token>";
        TableServiceClient serviceClientSas = new TableServiceClientBuilder()
                .endpoint(storageUri)
                .credential(new AzureSasCredential(sasToken))
                .buildClient();
        TableClient tableClientSas = serviceClientSas.getTableClient(tableName);
        System.out.println("Connected to Table Storage using SAS token. Table: " + tableName);

        // 3. Connect using Managed Identity (DefaultAzureCredential)
        // Make sure your environment supports Managed Identity (e.g., Azure VM, App Service, etc.)
        TableServiceClient serviceClientMsi = new TableServiceClientBuilder()
                .endpoint(storageUri)
                .credential(new DefaultAzureCredentialBuilder().build())
                .buildClient();
        TableClient tableClientMsi = serviceClientMsi.getTableClient(tableName);
        System.out.println("Connected to Table Storage using Managed Identity. Table: " + tableName);
    }
}
